parser_tool/static/js/src/spot_it.js

```diff
@@ -0,0 +1,482 @@
+      c = [0];
+      for (k = 0; k < n; k++) {
+        c.push(n + 1 + n * j + k);
+      }
+      deck.push(c);
+    }
+
+    //  for generating  n * n cards
+    for (i = 0; i < n; i++) {
+      for (j = 0; j < n; j++) {
+        c = [i + 1];
+        for (k = 0; k < n; k++) {
+          c.push(n + 1 + n * k + ((i * k + j) % n));
+        }
+        deck.push(c);
+      }
+    }
+    return deck;
+  };
+
+  const shuffle = (a) => {
+    const cloneA = [...a];
+    let j, x;
+    for (let idx = cloneA.length - 1; idx > 0; idx--) {
+      j = Math.floor(Math.random() * (idx + 1));
+      // swap - could use destructure [a[idx], a[j]] = [a[j], a[idx]]
+      x = cloneA[idx];
+      cloneA[idx] = cloneA[j];
+      cloneA[j] = x;
+    }
+    return cloneA;
+  };
+
+  const generateImageElement = (card, parent, cardIndex, data, pos) => {
+    let imgs = "";
+    for (let q = 0; q < card.length; q++) {
+      let idx = card[q];
+      let { id } = data[idx];
+      // TODO replace id with image
+      if (pos) {
+        let [right, top] = pos[q];
+        let size = generateRand(80, 160);
+        //TODO make random size fit on card without overlapping
+        imgs += `<button class="custom-button" style="position:relative; right:${right}px; top: ${top}px; transform:rotate(${generateRand(
+          0,
+          360
+        )}deg)" >
+							<img id="${idx}" style="min-width:${size}px;min-height:${size}px" value="${idx}" data-index="${cardIndex}" data-parent="${parent}" src=https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${id}.png />
+						</button>`;
+      } else {
+        imgs += `<button class="custom-button" >
+							<img id="${idx}" value="${idx}" data-index="${cardIndex}" data-parent="${parent}" src=https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${id}.png />
+						</button>`;
+      }
+    }
+    return imgs;
+  };
+
+  const generateRand = (a, b) => {
+    result = Math.floor(Math.random() * (b - a)) + a;
+    return result;
+  };
+
+  $(document).ready(async function () {
+    console.log("ready!");
+    const state = {
+      p1: "Player 1",
+      p2: "Player 2",
+      p1Score: 0,
+      p2Score: 0,
+      card1: {
+        value: [],
+        index: [],
+      },
+      card2: {
+        value: [],
+        index: [],
+      },
+      centralCard: {
+        value: [],
+        index: [],
+      },
+      foundPrimeNum: {},
+    };
+    // TODO ADD DECK AND CURRSELECTED TO STATE
+    let deck = [];
+    let currSelected = {};
+    let card1 = [];
+    let card2 = [];
+    let centralCard = [];
+    let deckElementP2 = "";
+    state.p1Score = 0;
+    state.p2Score = 0;
+    const data = JSON.parse(document.getElementById("data").textContent);
+    // static numbers to validate against data length
+    // image/button element position:[right,top]
+    // TODO make more dynamic - using static positions for images so they do not overlap
+    const maxNumReqImages = [
```

Comment:
Nice declarative structure for the possible game configurations (static is fine for now).